incubator/validator-avro/src/main/java/io/aklivity/zilla/runtime/validator/avro/AvroWriteValidator.java

```diff
@@ -141,11 +119,12 @@ private byte[] serializeJsonRecord(
         encoded.reset();
         try
         {
-            Schema schema = schemas.get(schemaId);
-            record = records.computeIfAbsent(schemaId, this::supplyRecord);
+            Schema schema = supplySchema(schemaId);
+            GenericDatumReader<GenericRecord> reader = supplyReader(schemaId);
+            GenericDatumWriter<GenericRecord> writer = supplyWriter(schemaId);
+            GenericRecord record = supplyRecord(schemaId);
```

Comment:
Can the json decoder below be precreated and reused instead of creating afresh each time?

Reply:
I think we can do that, however it requires to handle IOException so the AvroValidator constructor will have try/catch or throw exception handled. If that works, we can initialise this with a dummy schema & value.

Reply:
this doesn't seems possible as we don't have a method to take reuse JsonDecoder.

Reply:
but I was able to optimise BinaryEncoder & reuse the same instance.